banner/templatetags/banner_tags.py

```diff
@@ -1,43 +1,46 @@
+import six
+
 from django import template
 
+from banner.models import Banner
+from banner.styles import BANNER_STYLES_MAP
+
 register = template.Library()
 
 
 @register.tag
-def get_actual_banner(parser, token):
-    """Get actual banner for a BannerProxy object.
-
-    Syntax::
-
-        {% get_actual_banner for [object] as [varname] %}
-    """
-    tokens = token.contents.split()
-    if len(tokens) != 5:
-        raise template.TemplateSyntaxError(
-            "%r tag requires 4 arguments" % tokens[0]
-        )
+def render_banner(parser, token):
+    tokens = token.split_contents()
 
-    if tokens[1] != 'for':
+    if len(tokens) < 2:
         raise template.TemplateSyntaxError(
-            "First argument in %r tag must be 'for'" % tokens[0]
+            "render_banner tag requires at the slug or the banner object."
         )
 
-    if tokens[3] != 'as':
-        raise template.TemplateSyntaxError(
-            "Third argument in %r tag must be 'as'" % tokens[0]
-        )
+    object_or_slug = tokens[1]
+    kwargs = {}
+    for kv in tokens[2:]:
+        k, v = kv.split("=")
+        kwargs[k] = v
 
-    return ActualBannerNode(obj=tokens[2], as_var=tokens[4])
+    return BannerNode(object_or_slug, **kwargs)
 
 
-class ActualBannerNode(template.Node):
+class BannerNode(template.Node):
```

Comment:
Maybe rather look at something like:
```
class RenderBannerNode(template.Node):

    def __init__(self, object_or_slug, **kwargs):
        self.variable = template.Variable(object_or_slug)

    def render(self, context):
        try:
            variable = self.variable.resolve(context)
        except template.VariableDoesNotExist:
            variable = self.variable.var
        default_msg = "No Banner matches the given query. %s" % self.variable

        # If the variable is a string type, attempt to find object based on
        # slug field, otherwise pass the object along directly.
        if isinstance(variable, basestring):
            try:
                form = models.Banner.objects.get(slug=variable)
            except models.Banner.DoesNotExist:
                raise Http404(default_msg)
        elif isinstance(variable, models.Banner):
            banner = variable
        else:
            raise Http404(default_msg)
```

Reply:
:+1: